File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); ++i) {
            for (int j = 0; j < nums.size(); ++j) {
                if (i == j) continue;
                if (nums[i] + nums[j] == target) {
                    return std::vector<int>{i, j};
                }
            }
        }
        return std::vector<int>{-1, -1};
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-0.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        if nums.len() == 0 {
            panic!("empty array");
        }
        let mut cur_sum = 0;
        let mut max_sum = nums[0];
        for n in nums {
            cur_sum = max(cur_sum, 0) + n;
            max_sum = max(max_sum, cur_sum);
        }
        max_sum
    }
}